python/mxnet/symbol.py

```diff
@@ -832,14 +832,243 @@ def _get_ndarray_inputs(arg_key, args, arg_names, allow_missing):
+            grad_req_types = []
+            for name in listed_arguments:
+                if name in grad_req:
+                    grad_req_types.append(mx_uint(req_map[grad_req[name]]))
+                else:
+                    grad_req_types.append(mx_uint(0))
+
+        if group2ctx is not None:
+            if attr_dict is None:
+                attr_dict = self.attr_dict()
+            arg_ctx = [group2ctx.get(attr_dict[name]['__ctx_group__'], ctx)
+                       if name in attr_dict and '__ctx_group__' in attr_dict[name]
+                       else ctx for name in listed_arguments]
+            aux_ctx = [group2ctx.get(attr_dict[name]['__ctx_group__'], ctx)
+                       if name in attr_dict and '__ctx_group__' in attr_dict[name]
+                       else ctx for name in listed_aux_states]
+        else:
+            arg_ctx = [ctx] * len(listed_arguments)
+            aux_ctx = [ctx] * len(listed_aux_states)
+
+        ctx_map_keys = []
+        ctx_map_dev_types = []
+        ctx_map_dev_ids = []
+        if group2ctx is not None:
+            for key, val in group2ctx.items():
+                ctx_map_keys.append(c_str(key))
+                ctx_map_dev_types.append(ctypes.c_int(val.device_typeid))
+                ctx_map_dev_ids.append(ctypes.c_int(val.device_id))
+
+        # prepare param names
+        if param_names is None:
+            param_names = []
+        else:
+            if not isinstance(param_names, list):
+                raise ValueError('param_names in simple_bind must be a list or None')
+
+        # prepare shared_data_arrays
+        if shared_data_arrays is None:
+            num_shared_data_arrays = mx_uint()
+            shared_data_array_names = ctypes.POINTER(ctypes.c_char_p)()
+            shared_data_array_handles = ctypes.POINTER(NDArrayHandle)()
+        else:
+            if not isinstance(shared_data_arrays, dict):
+                raise ValueError('shared_data_arrays in simple_bind must be dict or None')
+            shared_data_array_names = []
+            shared_data_array_handles = []
+            for k, v in shared_data_arrays.items():
+                shared_data_array_names.append(k)
+                shared_data_array_handles.append(v.handle)
+            shared_data_array_names = ctypes.POINTER(ctypes.c_char_p)(shared_data_array_names)
+            num_shared_data_arrays = mx_uint(len(shared_data_array_handles))
+
+        # prepare shared_exec_in_args
+        if shared_exec is None:
+            num_shared_exec_in_args = 0
+            shared_exec_in_arg_handles = ctypes.POINTER(NDArrayHandle)()
+            num_shared_exec_arg_grads = 0
+            shared_exec_arg_grad_handles = ctypes.POINTER(NDArrayHandle)()
+            num_shared_exec_aux_states = 0
+            shared_exec_aux_state_handles = ctypes.POINTER(NDArrayHandle)()
+        else:
+            shared_exec_in_arg_handles = [nd.handle for nd in shared_exec.arg_arrays]
```

Comment:
I think you are right. GraphExecutor::data_entry_ has all the NDArrays of the full graph. In order to make use that, I want confirm that shared_exec's indexed graph is same as the current executor's indexed graph, right?